web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -1049,6 +1049,239 @@ $.extend(fixmystreet.set_up, {
         var set_map_state = true;
         fixmystreet.back_to_reports_list(e, report_list_url, map_state, set_map_state);
     });
+  },
+
+  expandable_list_items: function(){
+      // All of the .off/.on event binding here is intended to allow us
+      // to call fixmystreet.set_up.expandable_list_items() multiple times
+      // (eg: after report list items are loaded by ajax) without the
+      // effects stacking up on top of each other.
```

Comment:
But you're setting the event on document; so why do you ever need to set it up again once it's been set up the first time?

Reply:
Because I need to bind click handlers to the list items (which are grandparents of the `.js-toggle-expansion` elements), so I select them with `$('.js-toggle-expansion').closest('li')` on line 1071, and that means, if this gets run at pageload, I might need to run it again, later, if new `.js-toggle-expansion` elements are added to the DOM.

I guess we could avoid the `$('.js-toggle-expansion').closest('li');` line by binding to `$(document).on('click', '.item-list__item--expandable')` instead?

Reply:
Yes, something like that would be easier, I think (perhaps another js class for the item list to show it's a js binding class).